src/ripple/app/paths/impl/StrandFlow.h

```diff
@@ -543,15 +545,29 @@ flow (PaymentSandbox const& baseView,
         }
         if (!partialPayment)
         {
-            return {tecPATH_PARTIAL, actualIn, actualOut, std::move(ofrsToRmOnFail)};
+            // If we're offerCrossing a !partialPayment, then we're
+            // handling tfFillOrKill.  That case is handled below; not here.
+            if (!offerCrossing)
+                return {tecPATH_PARTIAL,
+                    actualIn, actualOut, std::move(ofrsToRmOnFail)};
         }
         else if (actualOut == beast::zero)
         {
             return {tecPATH_DRY, std::move(ofrsToRmOnFail)};
         }
     }
+    if (offerCrossing && !partialPayment)
+    {
+        // If we're offer crossing and partialPayment is *not* true, then
+        // we're handling a FillOrKill offer.  In this case remainingIn must
+        // be zero (all funds must be consumed) or else we kill the offer.
+        assert (remainingIn);
+        if (remainingIn && *remainingIn != zero)
```

Comment:
gcc warns here: `*** warning: ‘*((void*)(& remainingIn)+8).ripple::XRPAmount::drops_’ may be used uninitialized in this function [-Wmaybe-uninitialized]`